### PR TITLE
[SPARK-48720][SQL] Align the command `ALTER TABLE ... UNSET TBLPROPERTIES ...` in v1 and v2

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to:
- align the command `ALTER TABLE ... UNSET TBLPROPERTIES ...` in v1 and v2.
(this means that in the v1, regardless of whether `IF EXISTS` is specified or not, when unset a `non-existent` property, it is `ignored` and no longer `fails`.)
- update the description of `ALTER TABLE ... UNSET TBLPROPERTIES ...` in the doc `docs/sql-ref-syntax-ddl-alter-table.md`.
- unify v1 and v2 `ALTER TABLE ... UNSET TBLPROPERTIES ...` tests.
- Add the following `scenario` for `ALTER TABLE ... SET TBLPROPERTIES ...` testing
A.`table to alter does not exist`
B.`alter table set reserved properties`

### Why are the changes needed?
- align the command `ALTER TABLE ... UNSET TBLPROPERTIES ...` in v1 and v2, avoid confusing end-users.
- to improve test coverage.
- align with other similar tests, eg: `AlterTableSetTblProperties*`

### Does this PR introduce _any_ user-facing change?
Yes, in the `v1`, regardless of whether `IF EXISTS` is specified or not, when unset a `non-existent` property, it is `ignored` and no longer `fails`

### How was this patch tested?
Update some UT & Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
- In `Unify v1 and v2 ALTER TABLE .. UNSET TBLPROPERTIES tests` migration, I found syntax:
  ```
  ALTER TABLE ... UNSET TBLPROPERTIES IF EXISTS ('key1');
  ```
  v1: 
  ```
  If the property `key1` of the table do not exist, it will throw exception `UNSET_NONEXISTENT_PROPERTIES`
  ```
  v2 :

  ```
  If the property `key1` of the table do not exist,  it will ignore the property `key1` (non-existent) and finally succeed (slience)
  ```
- I originally planned to align its logic, which was to throw the same exception `UNSET_NONEXISTENT_PROPERTIES` as v1. However, based on the previous discussion in https://github.com/apache/spark/pull/47038, it seems that we can ·keep the current logic· and only reflect this difference in UT, while also `note` it in the document:

https://github.com/apache/spark/blob/baf461b5cb443e008059d3e8d6fa3d4971b9b698/docs/sql-ref-syntax-ddl-alter-table.md?plain=1#L244

@cloud-fan What do you think?

**Comment by cloud-fan** [comment]
Yea UNSET non-existing property should be noop. BTW we should real table not found error if table does not exist.

**Comment by panbingkun** [comment]
Okay, let me update it.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala [hunk 1]
```diff
@@ -1075,25 +1075,6 @@ class DDLParserSuite extends AnalysisTest {
         ifExists = true))
   }
 
-  // ALTER TABLE table_name UNSET TBLPROPERTIES [IF EXISTS] ('comment', 'key');
```
Comment (panbingkun) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala:1078]: The tests have been migrated to `AlterTableUnsetTblPropertiesParserSuite`

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala [hunk 2]
```diff
@@ -107,18 +107,6 @@ class DDLParserSuite extends AnalysisTest with SharedSparkSession {
         stop = 98))
   }
 
-  test("alter table unset properties - property values must NOT be set") {
```
Comment (panbingkun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala:110]: The test has been migrated to `AlterTableUnsetTblPropertiesParserSuite`

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 3]
```diff
@@ -1113,52 +1109,6 @@ abstract class DDLSuite extends QueryTest with DDLSuiteBase {
...
-        sql("ALTER TABLE tab1 UNSET TBLPROPERTIES ('c', 'xyz')")
-      },
-      errorClass = "UNSET_NONEXISTENT_PROPERTIES",
-      parameters = Map("properties" -> "`xyz`", "table" -> "`spark_catalog`.`dbx`.`tab1`")
-    )
-    // property to unset does not exist, but "IF EXISTS" is specified
-    sql("ALTER TABLE tab1 UNSET TBLPROPERTIES IF EXISTS ('c', 'xyz')")
-    assert(getProps == Map("x" -> "y"))
-  }
-
```
Comment (panbingkun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala:1161]: The above test was migrated to `AlterTableUnsetTblPropertiesSuite*` and split into `table to alter does not exist`, `alter table unset properties` and `alter table unset non-existent properties`, as `alter table unset non-existent properties` behaves differently in v1 and v2.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/AlterTableUnsetTblPropertiesSuite.scala [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      sql(s"CREATE TABLE $t (col1 int, col2 string, a int, b int) $defaultUsing")
+      val tableIdent = TableIdentifier("tbl", Some("ns"), Some(catalog))
+
+      sql(s"ALTER TABLE $t SET TBLPROPERTIES ('k1' = 'v1', 'k2' = 'v2', 'k3' = 'v3')")
+      checkTblProps(tableIdent, Map("k1" -> "v1", "k2" -> "v2", "k3" -> "v3"))
+
+      // property to unset does not exist
+      checkError(
+        exception = intercept[AnalysisException] {
+          sql(s"ALTER TABLE $t UNSET TBLPROPERTIES ('k3', 'k4')")
```
Comment (panbingkun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/AlterTableUnsetTblPropertiesSuite.scala:80]: This scenario uses different behaviors in `v1` and `v2`.

**Comment by panbingkun** [comment]
@cloud-fan It's ready now, thanks.

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 5]
```diff
@@ -236,20 +236,30 @@ ALTER TABLE table_identifier DROP [ IF EXISTS ] partition_spec [PURGE]
...
 ALTER TABLE table_identifier SET TBLPROPERTIES ( key1 = val1, key2 = val2, ... )
+```
+
+#### UNSET PROPERTIES
+
+`ALTER TABLE UNSET` command is used to drop the table property.
 
--- Unset Table Properties
+**Note:** If the specified property key does not exist, when you use the v1 command and do not specify `IF EXISTS`,
+it will throw the error-condition `UNSET_NONEXISTENT_PROPERTIES` and finally `failed`,
```
Comment (cloud-fan) [inline docs/sql-ref-syntax-ddl-alter-table.md:256]: shall we change the v1 behavior as well? It's not a breaking change to change from failure to no failure.

Comment (cloud-fan) [reply docs/sql-ref-syntax-ddl-alter-table.md:256]: and we should avoid mentioning v1/v2 in the doc. It's an internal concept and should not be exposed to end users

Comment (panbingkun) [reply docs/sql-ref-syntax-ddl-alter-table.md:256]: > shall we change the v1 behavior as well? It's not a breaking change to change from failure to no failure.



> shall we change the v1 behavior as well? It's not a breaking change to change from failure to no failure.

Sure, let me update it.

Comment (panbingkun) [reply docs/sql-ref-syntax-ddl-alter-table.md:256]: > and we should avoid mentioning v1/v2 in the doc. It's an internal concept and should not be exposed to end users

Okay.

Comment (panbingkun) [reply docs/sql-ref-syntax-ddl-alter-table.md]: > shall we change the v1 behavior as well? It's not a breaking change to change from failure to no failure.

Done.

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 6]
```diff
@@ -236,20 +236,28 @@ ALTER TABLE table_identifier DROP [ IF EXISTS ] partition_spec [PURGE]
...
+`ALTER TABLE UNSET` command is used to drop the table property.
 
--- Unset Table Properties
+**Note:** If the specified property key does not exist, whether specify `IF EXISTS` or not, the command will ignore it and finally succeed.
+
+##### Syntax
+
+```sql
+-- Unset Properties
 ALTER TABLE table_identifier UNSET TBLPROPERTIES [ IF EXISTS ] ( key1, key2, ... )
```
Comment (cloud-fan) [inline docs/sql-ref-syntax-ddl-alter-table.md:261]: let's remove `[ IF EXISTS ]` as a way to deprecate. It's no longer effective.

**Review by cloud-fan — APPROVED** [review]

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 6 above)
Comment (panbingkun) [reply docs/sql-ref-syntax-ddl-alter-table.md]: Done.

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Pull request closed without merging**